File: spec/java/src/io/kaitai/struct/spec/TestStrEncodingsDefault.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.StrEncodingsDefault;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestStrEncodingsDefault extends CommonSpec {
    @Test
    public void testStrEncodingsDefault() throws Exception {
        StrEncodingsDefault r = StrEncodingsDefault.fromFile(SRC_DIR + "str_encodings.bin");

        assertEquals(r.str1(), "Some ASCII");
        assertEquals(r.rest().str2(), "こんにちは");
        assertEquals(r.rest().str3(), "こんにちは");
        assertEquals(r.rest().str4(), "░▒▓");
    }
}

File: spec/javascript/test_str_encodings_default.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('StrEncodingsDefault', 'src/str_encodings.bin', function(r) {
  assert.equal(r.str1, 'Some ASCII');
  assert.equal(r.rest.str2, 'こんにちは');
  assert.equal(r.rest.str3, 'こんにちは');
  assert.equal(r.rest.str4, '░▒▓');
});
